protocol/test/foundry/PipelineConvert.t.sol

```diff
@@ -0,0 +1,1461 @@
+        assertTrue(overallDeltaB != 0);
+    }
+
+
+    function testDeltaBChangeBeanToLP(uint256 amount) public {
+        amount = bound(amount, 1e6, 5000e6);
+        int256 beforeDeltaB = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        
+        doBasicBeanToLP(amount, users[1]);
+
+        int256 afterDeltaB = seasonGetters.poolDeltaBInsta(C.BEAN_ETH_WELL);
+        assertTrue(afterDeltaB < beforeDeltaB);
+        assertTrue(beforeDeltaB - int256(amount)*2 < afterDeltaB);
+        // would be great to calcuate exactly what the new deltaB should be after convert
+    }
+
+    function testTotalStalkAmountDidNotIncrease(uint256 amount) public {
+        amount = bound(amount, 1e6, 5000e6);
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+        uint256 beforeTotalStalk = bs.totalStalk();
+        beanToLPDoConvert(amount, stem, users[1]);
+
+        uint256 afterTotalStalk = bs.totalStalk();
+        assertTrue(afterTotalStalk <= beforeTotalStalk);
+    }
+
+    function testUserStalkAmountDidNotIncrease(uint256 amount) public {
+        amount = bound(amount, 1e6, 5000e6);
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+        uint256 beforeUserStalk = bs.balanceOfStalk(users[1]);
+        beanToLPDoConvert(amount, stem, users[1]);
+
+        uint256 afterUserStalk = bs.balanceOfStalk(users[1]);
+        assertTrue(afterUserStalk <= beforeUserStalk);
+    }
+
+    function testUserBDVDidNotIncrease(uint256 amount) public {
+        amount = bound(amount, 1e6, 5000e6);
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+        uint256 beforeUserDeposit = bs.balanceOfDepositedBdv(users[1], C.BEAN);
+        beanToLPDoConvert(amount, stem, users[1]);
+
+        uint256 afterUserDeposit = bs.balanceOfDepositedBdv(users[1], C.BEAN);
+        assertTrue(afterUserDeposit <= beforeUserDeposit);
+    }
+
+    function testConvertAgainstPegAndLoseStalk(uint256 amount) public {
+        amount = bound(amount, 5000e6, 5000e6); // todo: update for range
+        
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+        // uint256 beforeTotalStalk = bs.totalStalk();
+        uint256 grownStalkBefore = bs.balanceOfGrownStalk(users[1], C.BEAN);
+
+
+        beanToLPDoConvert(amount, stem, users[1]);
+
+        uint256 grownStalkAfter = bs.balanceOfGrownStalk(users[1], C.BEAN_ETH_WELL);
+
+        assertTrue(grownStalkAfter == 0); // all grown stalk was lost
+        assertTrue(grownStalkBefore > 0);
+    }
+
+    function testConvertWithPegAndKeepStalk(uint256 amount) public {
+        amount = bound(amount, 5000e6, 5000e6); // todo: update for range
+
+        // how many eth would we get if we swapped this amount in the well
+        uint256 ethAmount = IWell(C.BEAN_ETH_WELL).getSwapOut(IERC20(C.BEAN), IERC20(C.WETH), amount);
+        ethAmount = ethAmount.mul(2); // I need a better way to calculate how much eth out there should be to make sure we can swap and be over peg
```

Comment:
Can you explain this?

Reply:
Didn't have a good way to set deltaB at the time, fixed with latest deltaB setting function: https://github.com/BeanstalkFarms/Beanstalk/commit/fc40f2cead4770eadcc008bbe5cbcb76525b5295